portality/static/js/formulaic.js

```diff
@@ -1195,10 +1195,30 @@ var formulaic = {
 
             this.init();
         },
-
         newClickableOwner : function(params) {
             return edges.instantiate(formulaic.widgets.ClickableOwner, params)
         },
+        newClickToCopy : function(params) {
+            return edges.instantiate(formulaic.widgets.ClickToCopy, params)
+        },
+        ClickToCopy : function(params) {
+            this.fieldDef = params.fieldDef;
+            this.init = function() {
+                var elements = $("#click-to-copy--" + this.fieldDef.name);
+                edges.on(elements, "click", this, "copy");
+            };
+            this.copy = function(element) {
+                let form = new doaj.af.BaseApplicationForm()
+                let value = form.determineFieldsValue(this.fieldDef.name)
+                let value_to_copy = form.convertValueToText(value);
+                navigator.clipboard.writeText(value_to_copy)
+                var confirmation = $("#copy-confirmation--" + this.fieldDef.name);
+                confirmation.text("Value copied: " + value_to_copy);
+                confirmation.show().delay(3000).fadeOut();
+            };
+            this.init();
+
+        },
         ClickableOwner : function(params) {
             this.fieldDef = params.fieldDef;
```

Comment:
suggest add function `newClickToCopy` and `ClickToCopy` after `ClickableOwner`
because 
`newClickableOwner`, `ClickableOwner` should be a pair of functions.

the current order is 
```
newClickableOwner
newClickToCopy
ClickToCopy
ClickableOwner
```

recommended order is 
```
newClickableOwner
ClickableOwner
newClickToCopy
ClickToCopy
```